### PR TITLE
Updated NEWS file and version numbers for v1.5.0rc1

### PR DESCRIPTION
**Pull request by rdesai16**

### CODE REVIEW

**Comment by lanl-ompi** [comment]
Can one of the admins verify this patch?

**Comment by rdesai16** [comment]
@davidozog @wrrobin I've made the changes as per the release check list. Let me know if you find something missing here, especially in the NEWS file.

File: NEWS [hunk 1]
```diff
@@ -2,6 +2,13 @@
 
 Sandia OpenSHMEM NEWS -- history of user-visible changes.
 
+v1.5.0rc1
+---------
+- Renamed all OpenSHMEM v1.5 interfaces from "shmemx" to "shmem" and moved all unit tests accordingly
```
Comment (davidozog) [inline NEWS:7]: Perhaps we should re-word this to match our previous release notes.  Something like:
```
- Support for OpenSHMEM v1.5 specification
- New features include: a teams API and teams-based collectives, put-with-signal routines, nonblocking atomic routines, multiple-element wait/test vector comparison routines, shmem_malloc_with_hints, and a profiling interface.  See OpenSHMEM 1.5 Specification, Annex G for details.
- Deprecations include: active-set-based library constants and collective routines, shmem_barrier, and short/ unsigned short variants for shmem_wait_until and shmem_test.  See OpenSHMEM 1.5 Specification, Annex F for details.
```

File: NEWS [hunk 2]
```diff
@@ -2,6 +2,13 @@
...
+- Added support for UCX Transport
```
Comment (davidozog) [inline NEWS:8]: Lower-case `transport`.

File: NEWS [hunk 3]
```diff
@@ -2,6 +2,13 @@
...
+- Added shmem_malloc_with_hints(though no hints supported yet)
```
Comment (davidozog) [inline NEWS:9]: please add a space before the parens

Comment (wrrobin) [reply NEWS:9]: Is it better to re-write this? 
`- Added shmem_malloc_with_hints. Currently, no hint values are supported.`

**Pull request merged**